CMakeLists.txt

```diff
@@ -1588,15 +1596,31 @@ target_include_directories(libnetdata BEFORE PUBLIC ${LIBUV_INCLUDE_DIRS})
 target_compile_definitions(libnetdata PUBLIC ${LIBUV_CFLAGS_OTHER})
 target_link_libraries(libnetdata PUBLIC ${LIBUV_LDFLAGS})
 
-# crypto
-target_include_directories(libnetdata BEFORE PUBLIC ${CRYPTO_INCLUDE_DIRS})
-target_compile_options(libnetdata PUBLIC ${CRYPTO_CFLAGS_OTHER})
-target_link_libraries(libnetdata PUBLIC ${CRYPTO_LDFLAGS})
+if (ENABLE_OPENSSL)
+        message(STATUS "Compiling Netdata with OpenSSL")
+        # crypto
+        target_include_directories(libnetdata BEFORE PUBLIC ${CRYPTO_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${CRYPTO_CFLAGS_OTHER})
+        target_link_libraries(libnetdata PUBLIC ${CRYPTO_LDFLAGS})
 
-# openssl
-target_include_directories(libnetdata BEFORE PUBLIC ${OPENSSL_INCLUDE_DIRS})
-target_compile_options(libnetdata PUBLIC ${OPENSSL_CFLAGS_OTHER})
-target_link_libraries(libnetdata PUBLIC ${OPENSSL_LDFLAGS})
+        # openssl
+        target_include_directories(libnetdata BEFORE PUBLIC ${OPENSSL_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${OPENSSL_CFLAGS_OTHER})
+        target_link_libraries(libnetdata PUBLIC ${OPENSSL_LDFLAGS})
+endif()
+
+if (ENABLE_WOLFSSL)
+        message(STATUS "Compiling Netdata with WolfSSL")
+
+        target_include_directories(libnetdata BEFORE PUBLIC ${WOLFSSL_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${WOLFSSL_CFLAGS_OTHER})
+        target_link_libraries(libnetdata PUBLIC ${WOLFSSL_LDFLAGS})
+
+        check_function_exists(wolfSSL_set_alpn_protos HAVE_WOLFSSL_SET_ALPN_PROTOS)
+        if(NOT HAVE_WOLFSSL_SET_ALPN_PROTOS)
+                message(FATAL_ERROR "WolfSSl libray is not compiled with all necessary functions.")
+        endif()
```

Comment:
This needs to be up at the top where we’re checking for the availability of WolfSSL, because it’s an integral part of us checking for a usable copy of WolfSSL.

Reply:
I moved it with latest commits;